peg/kotlin.peg

```diff
@@ -22,6 +22,10 @@
 
```

Comment:
I've made few more adjustments to the grammar. Could you please use the file from 
https://github.com/dolik-rce/ctags/blob/35ea29c889b345946b3cd11bf3593b82de2de4eb/peg/kotlin.peg?

Reply:
I put the new file to "build-sys: use upstream packcc" commit.
Is this o.k.?

We should separate the changes about kotlin.peg to another commit. However, when I removed the change about kotlin.peg from "build-sys: use upstream packcc" commit, I cannot build ctags. So we may have to put everything into the commit.

Reply:
packcc used in u-ctags was derived from enechaev/packcc.
I wonder the fixes in https://github.com/enechaev/packcc/commits/master are in the new upstream.

Reply:
> I put the new file to "build-sys: use upstream packcc" commit.
Is this o.k.?

I don't mind. If you wan't it in two separate commits and both of them passing tests, then you'd  have to first change the grammar and then add `%earlysource` later in the second commit.

> I wonder the fixes in https://github.com/enechaev/packcc/commits/master are in the new upstream.

If you mean those 6 changes you listed in arithy/packcc#25, then yes, those are present in the new upstream. I'm not sure about other changes, but I think it was mostly features added by enechaev. As far as I can tell, everything works just fine with current state of this branch (I just checked again now).

Reply:
I made a separate commit for changing about kotlin.peg. Thank you.